Sources/PackageModel/UserToolchain.swift

```diff
@@ -128,7 +129,12 @@ public final class UserToolchain: Toolchain {
                                              environment: environment) {
                     return librarian.basename
                 }
-                // TODO(5719) use `lld-link` if the build requests lld.
+                // TODO(5719) handle `-Xmanifest` vs `-Xswiftc`.
+                // `-use-ld=` is always joined in Swift.
+                if let ld = extraSwiftFlags.first(where: { $0.starts(with: "-use-ld=") }) {
```

Comment:
Seems a bit unfortunate to have to parse the flags like this, can/should we ask users to configure alternative linkers more explicitly?

Reply:
I agree it is unfortunate.  How would you recommend that we set that up?  The general mechanism for this is `-use-ld=...` which with CMake you would pass via `CMAKE_Swift_FLAGS` and that is honoured everywhere.

Reply:
if we have to parse, could we use argument parser to help make this more reliable?

Reply:
I'm not sure if that is going to be any better because we will need to consider the case of `-Xswiftc` or `-Xclang-linker` where the option itself is made opaque.  We would need to process both the driver and frontend flags and possibly non-swift-tool flags.